From 72058f9253e7ed8c7243e2ff76a16d97b03d65ed Mon Sep 17 00:00:00 2001
From: Jake Hunsaker <jhunsake@redhat.com>
Date: Tue, 25 Sep 2018 12:05:59 -0400
Subject: [PATCH] [sos_collector] Set umask for all files created

Sets a umask of 0077 for all files created by sos-collector. This
ensures collected sosrpeorts are not world readable and maintains the
original permissions of the sos archives.

Signed-off-by: Jake Hunsaker <jhunsake@redhat.com>
---
 soscollector/sos_collector.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/soscollector/sos_collector.py b/soscollector/sos_collector.py
index 874022b..f1d4716 100644
--- a/soscollector/sos_collector.py
+++ b/soscollector/sos_collector.py
@@ -41,6 +41,7 @@ class SosCollector():
     '''Main sos-collector class'''
 
     def __init__(self, config):
+        os.umask(0077)
         self.config = config
         self.threads = []
         self.workers = []
